### PR TITLE
Allow Public and IDP submitter access regardless of roles.

### PR DESCRIPTION
**Pull request by parc-jason**
See [SHOWCASE-1730](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-1730)

The issue was that any role assignment was considered more important than the Public or IDP access.  If a form is Public, the user should always have submission permissions ({submission_create,form_read}).  If a form is protected by a particular IDP and the use has that IDP, they should always have submission permissions ({submission_create,form_read}).  It shouldn't matter if they have other roles (which may NOT include submission permissions).
Basically, make public/IDP permissions additive to role permissions.

Also... Bugfix read for permission and role.  D'oh.

# Description

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-wxpbtr-tools.pathfinder.gov.bc.ca/dashboard?id=common-hosted-form-service-pr-87

**Review by loneil — CHANGES_REQUESTED** [review]
Still get unauthorized on the case I'm testing...
(note the Launch button doesn't show up in the IDIR-but-with-roles test case and that's a separate UI issue, but the submission should work directly navigating)

Looking into it more. Form a5746ecb-050e-40d4-8b2a-fa60b4ef0a0c on the PR and CSSTROL1 IDIR user.

**Comment by loneil** [comment]
@parc-jason debugged this PR locally, not sure the root cause.
Created a form with IDIR permission, managed role for CSSTROL1, didn't touch CSSTROL2. First one can't submit the form.

For CSSTROL1, I've granted them an additional role (submitter)
line 66 of `middleware\userAccess.js`
```
intersection -> [ 'form_read' ]
permissions -> [ 'form_read', 'submission_create' ]
```
This fails with a 401 trying to post the form

For CSSTROL2, no changes to that IDIRs role on the form
```
intersection -> [ 'form_read', 'submission_create' ]
permissions -> [ 'form_read', 'submission_create' ]
```
So that succeeds posting the form.

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-wxpbtr-tools.pathfinder.gov.bc.ca/dashboard?id=common-hosted-form-service-pr-87
